## Ćwiczenie 1: Series — tworzenie i operacje 

In [1]:
import pandas as pd
import numpy as np
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)


0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


In [ ]:
#1a Tworzenie Series pensje ze słownika

# Klucze: 'Anna', 'Jan', 'Ewa', 'Marek', 'Kasia'
# Wartości: 5500, 7200, 4800, 9100, 6300
pensje = pd.Series({
    'Anna' : 5500, 
    'Jan' : 7200, 
    'Ewa' : 4800, 
    'Marek' : 9100, 
    'Kasia' : 6300
    })
print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


In [8]:
#1b Pytania o pensję

# 1. Wyświetl imię osoby zarabiającej najwięcej (użyj idxmax())
najwiecej = pensje.idxmax()
print(najwiecej)

# 2. Oblicz średnią pensję
srednia = pensje.mean()
print(srednia)

# 3. Wyświetl imiona osób zarabiających powyżej średniej (filtrowanie boolean)

powyzej_sredniej = pensje[pensje > srednia]
print(powyzej_sredniej)

# 4. Policz ile osób zarabia poniżej 6000 (użyj len() lub .sum())
ponizej_6000 = pensje < 6000

liczba_osob = ponizej_6000.sum()
print(liczba_osob)

Marek
6580.0
Jan      7200
Marek    9100
dtype: int64
2


In [16]:
#1c Pensja po zmianach

# 1. Oblicz pensje po podwyżce 10% dla wszystkich (pensje * 1.10)
pensja_podwyżka = pensje * 1.10

# 2. Oblicz pensje po premii 500 zł dla wszystkich (pensje + 500)
pensja_premia = pensje + 500

# 3. Oblicz nową pensję = (pensja * 1.10) + 500
nowa_pensja = (pensje * 1.10) + 500

# 4. Wyświetl ranking nowych pensji od najwyższej (użyj sort_values)
pensje_posortowane = nowa_pensja.sort_values(ascending=False)
print(pensje_posortowane)

Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


## Ćwiczenie 2: DataFrame — tworzenie i atrybuty

In [17]:
# Dane o produktach sklepu internetowego
sklep = pd.DataFrame({
'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})


In [18]:
#2a Eksploracja DataFrame sklep

# 1. Wyświetl kształt tabeli (shape)
print(sklep.shape)

# 2. Wyświetl nazwy kolumn (columns)
print(sklep.columns)

# 3. Wyświetl typy danych w każdej kolumnie (dtypes)
print(sklep.dtypes)

# 4. Wyświetl pierwsze 3 wiersze (head)
print(sklep.head(3))

(5, 4)
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


In [19]:
#2b Dodawanie nowych kolumn do DataFrame sklep

# 1. Dodaj kolumnę 'wartosc_magazynu' = cena * stan_magazynowy
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']

# 2. Dodaj kolumnę 'cena_z_vat' = cena * 1.23
sklep['cena_z_vat'] = sklep['cena'] * 1.23

# 3. Wyświetl cały DataFrame z nowymi kolumnami
print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [20]:
#2c Eksploracja datasetu tips (CSV)

# Wczytaj dataset tips
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

# 1. Wyświetl ile wierszy i kolumn ma dataset (shape)
print(tips.shape)

# 2. Wyświetl ostatnie 5 wierszy (tail)
print(tips.tail())

# 3. Wyświetl 3 losowe wiersze (sample)
print(tips.sample(3))

# 4. Wyświetl typy danych w każdej kolumnie (dtypes)
print(tips.dtypes)

(244, 7)
     total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2
     total_bill   tip     sex smoker   day   time  size
133       12.26  2.00  Female     No  Thur  Lunch     2
127       14.52  2.00  Female     No  Thur  Lunch     2
121       13.42  1.68  Female     No  Thur  Lunch     2
total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


## Ćwiczenie 3: Eksploracja datasetu — samodzielna analiza

In [21]:
#3a Pierwszy rzut oka

url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

# 1. Wyświetl pierwsze 10 wierszy (head)
print(penguins.head(10))

# 2. Wyświetl kształt tabeli (shape)
print(penguins.shape)

# 3. Uruchom info() — ile brakujących wartości widzisz?
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

In [22]:
#3b Statystyki opisowe

# 1. Uruchom describe() — statystyki numeryczne
print(penguins.describe())

# 2. Uruchom describe(include='all') — co dodają kolumny tekstowe?
print(penguins.describe(include='all'))

# 3. Mediana masy ciała (body_mass_g)
mediana_masa = penguins['body_mass_g'].median()
print("Mediana body_mass_g:", mediana_masa)

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000
       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   

In [23]:
#3c Rozkłady wartości

# 1. Ile pingwinów każdego gatunku? (species)
print(penguins['species'].value_counts())

# 2. Ile pingwinów na każdej wyspie? (island)
print(penguins['island'].value_counts())

# 3. Rozkład płci (sex) — czy jest zbalansowany?
print(penguins['sex'].value_counts())

# 4. Jaki procent to gatunek Adelie? (normalize=True)
print(penguins['species'].value_counts(normalize=True) * 100)

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64
sex
MALE      168
FEMALE    165
Name: count, dtype: int64
species
Adelie       44.186047
Gentoo       36.046512
Chinstrap    19.767442
Name: proportion, dtype: float64


In [24]:
#3d Brakujące dane

# 1. Policz brakujące wartości w każdej kolumnie
print(penguins.isna().sum())

# 2. Jaki procent danych brakuje w kolumnie 'sex'?
braki_sex = penguins['sex'].isna().sum()
procent_brak = braki_sex / len(penguins) * 100
print("Procent braków w kolumnie 'sex':", procent_brak, "%")

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64
Procent braków w kolumnie 'sex': 3.1976744186046515 %


#3e Wnioski

1. Dataset zawiera 344 pingwinów z 3 gatunków.
2. Najczęstszy gatunek to Adelie (152 osobników, 44.2%).
3. Najmniej liczny gatunek to Chinstrap (68 osobników, 19.8%).
4. Brakuje danych w kolumnie 'sex' — 11 braków (3.2%).
5. Średnia masa ciała to około 4201 g (mediana: 4050 g).
6. Pingwiny mieszkają na 3 wyspach, najliczniej na wyspie Biscoe (168 osobników).

## Ćwiczenie 4: Pytania biznesowe + commit

In [25]:
#4a Pytania biznesowe — dataset tips

import pandas as pd

# Wczytanie datasetu
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')

# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

In [26]:
#1. Oblicz średni rachunek (total_bill)
sredni_rachunek = tips['total_bill'].mean()
print("Średni rachunek:", round(sredni_rachunek, 2))

Średni rachunek: 19.79


In [27]:
#2. Oblicz średni napiwek w procentach (tip_pct)
sredni_tip_pct = tips['tip_pct'].mean()
print("Średni napiwek (%):", round(sredni_tip_pct, 1))

Średni napiwek (%): 16.1


In [28]:
#3. Wyświetl który dzień ma najwięcej rachunków (value_counts)
najwiecej_dzien = tips['day'].value_counts().idxmax()
print("Dzień z największą liczbą rachunków:", najwiecej_dzien)

Dzień z największą liczbą rachunków: Sat


In [29]:
#4. Policz ile rachunków jest od palaczy (value_counts na 'smoker')
rachunki_palacze = tips['smoker'].value_counts()
print("Rachunki od palaczy i niepalących:\n", rachunki_palacze)

Rachunki od palaczy i niepalących:
 smoker
No     151
Yes     93
Name: count, dtype: int64


In [30]:
#5. Wyświetl describe() tylko dla kolumny 'tip'
print(tips['tip'].describe())

# Mediana napiwku
mediana_tip = tips['tip'].median()
print("Mediana napiwku:", mediana_tip)

count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64
Mediana napiwku: 2.9


In [31]:
#6. Jaki procent rachunków pochodzi z obiadu (Dinner)?
procent_dinner = tips['time'].value_counts(normalize=True)['Dinner'] * 100
print("Procent rachunków Dinner:", round(procent_dinner, 1), "%")

Procent rachunków Dinner: 72.1 %


In [32]:
#7. Ile unikalnych rozmiarów grup jest w danych? (nunique na 'size')
unikalne_rozmiary = tips['size'].nunique()
print("Liczba unikalnych rozmiarów grup:", unikalne_rozmiary)

Liczba unikalnych rozmiarów grup: 6
